### ISSUE TITLE
logclean command removes all logs instead of only logs older than X days

### ISSUE THREAD

**Issue by peikk0** [freebsd/poudriere#571]
Trying to remove all logs older than 15 days:
```
% date
Wed 27 Dec 2017 15:46:13 NZDT
% sudo poudriere logclean -j releng_11_1 -n 15
[00:00:00] Looking for builds older than 15 days in /usr/local/poudriere/data/logs/bulk (filtered)... done
[00:00:00] Calculating size for found files... done
[00:00:00] These builds older than 15 days in /usr/local/poudriere/data/logs/bulk (filtered) will be deleted:
releng_11_1-default-server/2017-11-29_10h30m16s
releng_11_1-default-server/2017-11-30_08h06m08s
releng_11_1-default-server/2017-11-30_15h02m00s
releng_11_1-default-server/2017-11-30_15h07m59s
releng_11_1-default-server/2017-12-01_08h05m42s
releng_11_1-default-server/2017-12-02_08h05m18s
releng_11_1-default-server/2017-12-02_13h29m47s
releng_11_1-default-server/2017-12-02_13h36m31s
releng_11_1-default-server/2017-12-02_21h14m08s
releng_11_1-default-server/2017-12-02_21h20m04s
releng_11_1-default-server/2017-12-02_22h10m43s
releng_11_1-default-server/2017-12-02_22h11m49s
releng_11_1-default-server/2017-12-03_08h05m12s
releng_11_1-default-server/2017-12-03_11h59m33s
releng_11_1-default-server/2017-12-04_08h05m16s
releng_11_1-default-server/2017-12-05_08h05m15s
releng_11_1-default-server/2017-12-06_01h47m20s
releng_11_1-default-server/2017-12-06_08h05m14s
releng_11_1-default-server/2017-12-07_08h05m16s
releng_11_1-default-server/2017-12-08_08h05m46s
releng_11_1-default-server/2017-12-09_08h05m16s
releng_11_1-default-server/2017-12-10_08h05m21s
releng_11_1-default-server/2017-12-10_22h26m14s
releng_11_1-default-server/2017-12-11_08h05m13s
releng_11_1-default-server/2017-12-12_08h05m19s
releng_11_1-default-server/2017-12-13_08h05m23s
releng_11_1-default-server/2017-12-14_08h05m15s
releng_11_1-default-server/2017-12-15_08h05m18s
releng_11_1-default-server/2017-12-15_13h03m09s
releng_11_1-default-server/2017-12-16_08h05m16s
releng_11_1-default-server/2017-12-17_08h05m14s
releng_11_1-default-server/2017-12-18_08h05m15s
releng_11_1-default-server/2017-12-19_08h05m49s
releng_11_1-default-server/2017-12-20_08h05m16s
releng_11_1-default-server/2017-12-21_08h05m16s
releng_11_1-default-server/2017-12-22_08h05m16s
releng_11_1-default-server/2017-12-23_08h05m16s
releng_11_1-default-server/2017-12-24_08h05m19s
releng_11_1-default-server/2017-12-25_08h05m16s
releng_11_1-default-server/2017-12-26_08h05m16s
releng_11_1-default-server/2017-12-27_08h05m16s
[00:00:00] Removing these builds older than 15 days in /usr/local/poudriere/data/logs/bulk (filtered) will free: 67.78 MiB
[00:00:00] Dry run: not cleaning anything.
[...]
```
I've tried running `find $logs -type d -mindepth 0 -maxdepth 0 -Btime +15d` manually and it lists everything just like poudriere above, could that be an issue related to ZFS and inode creation time?

**Comment by bdrewery** [comment]
Which kernel are you running? That's really odd. Was this directory snapshot rolled back or received?

I don't get the same behavior locally on 12.0-CURRENT #7 r324671 with ZFS.

**Comment by peikk0** [comment]
I'm running 11.1-RELEASE-p4. The volume hasn't been rolled back or anything. I have atime=off on the entire zpool, could that be it?

**Comment by bdrewery** [comment]
No

**Comment by koobs** [comment]
I'm seeing the same symptoms but for `logclean -N10` (count) command:

```
poudriere-devel-3.3.99.20200326_2  =   up-to-date with index
```

```
[koobs@130-CURRENT-amd64:/usr/ports] ls /storage/poudriere/data/logs/bulk/12amd64-koobs/2020-12-04_0<tab>
2020-12-04_03h12m51s/ 
2020-12-04_03h28m22s/ 
2020-12-04_03h30m32s/ 
2020-12-04_03h32m57s/
2020-12-04_05h12m45s/
2020-12-04_05h18m51s/
```

Output

```
[koobs@130-CURRENT-amd64:/usr/ports] poudriere logclean -p koobs -n -N 10 |grep 2020-12
... snip ...
12amd64-koobs/2020-12-04_03h12m51s
12amd64-koobs/2020-12-04_03h28m22s
12amd64-koobs/2020-12-04_03h30m32s
12amd64-koobs/2020-12-04_03h32m57s
12amd64-koobs/2020-12-04_05h12m45s
12amd64-koobs/2020-12-04_05h18m51s
```

I don't expect any 2020-12-04 directories to be deleted

I see the same output **omitting** `-p koobs` as well, which is even more surprising, unless not specifying `-p` means "all ports trees", which would appear inconsistent with the logclean usage:

```
    -p tree     -- Specify which ports tree to use for log directories
                   (Defaults to the 'default' tree)
```

[koobs@130-CURRENT-amd64:/usr/ports] poudriere logclean -n -N 10 |grep 2020-12
12amd64-koobs/2020-12-04_03h12m51s
12amd64-koobs/2020-12-04_03h28m22s
12amd64-koobs/2020-12-04_03h30m32s
12amd64-koobs/2020-12-04_03h32m57s
12amd64-koobs/2020-12-04_05h12m45s
12amd64-koobs/2020-12-04_05h18m51s

**Comment by bdrewery** [comment]
Koobs issue is not the same as this; that's using `-N and is addressed in #805`. This bug is about `<days>`.

As noted earlier this appears to not be a bug in Poudriere.
> I've tried running find $logs -type d -mindepth 0 -maxdepth 0 -Btime +15d manually and it lists everything just like poudriere above, could that be an issue related to ZFS and inode creation time?

I am unable to reproduce the problem with `find` nor `logclean`.

```
~/git/poudriere6 # find /poudriere/data/logs/bulk/exp-head-i386-commit-test/*-*-*_*s -mindepth 0 -maxdepth 0 -type d -exec stat {} +
5405910403312123072 163848 drwxr-xr-x 4 root wheel 18446744073709551615 65 "Feb  9 16:41:24 2018" "Mar 23 12:56:29 2021" "Mar 23 12:56:29 2021" "Apr 20 14:21:54 2018" 4608 17 0x800 /poudriere/data/logs/bulk/exp-head-i386-commit-test/2018-04-20_14h21m43s
5405910403312123072 493897 drwxr-xr-x 4 root wheel 18446744073709551615 64 "Feb  9 16:41:24 2018" "Jul  9 14:24:10 2018" "Jul  9 14:24:10 2018" "Jul  9 13:41:54 2018" 4608 17 0x800 /poudriere/data/logs/bulk/exp-head-i386-commit-test/2018-07-09_13h41m16s
5405910403312123072 503898 drwxr-xr-x 4 root wheel 18446744073709551615 57 "Feb  9 16:41:24 2018" "Jul  9 18:35:40 2018" "Jul  9 18:35:40 2018" "Jul  9 14:24:28 2018" 4096 17 0x800 /poudriere/data/logs/bulk/exp-head-i386-commit-test/2018-07-09_14h24m21s
5405910403312123072 558060 drwxr-xr-x 4 root wheel 18446744073709551615 39 "Feb  9 16:41:24 2018" "Jul 10 14:06:54 2018" "Jul 10 14:06:54 2018" "Jul 10 14:06:43 2018" 4096 17 0x800 /poudriere/data/logs/bulk/exp-head-i386-commit-test/2018-07-10_14h06m35s
5405910403312123072 558154 drwxr-xr-x 4 root wheel 18446744073709551615 39 "Feb  9 16:41:24 2018" "Jul 10 16:23:25 2018" "Jul 10 16:23:25 2018" "Jul 10 14:28:52 2018" 4096 17 0x800 /poudriere/data/logs/bulk/exp-head-i386-commit-test/2018-07-10_14h28m45s
5405910403312123072 576523 drwxr-xr-x 4 root wheel 18446744073709551615 35 "Feb  9 16:41:24 2018" "Jul 10 23:00:46 2018" "Jul 10 23:00:46 2018" "Jul 10 23:00:39 2018" 4096 17 0x800 /poudriere/data/logs/bulk/exp-head-i386-commit-test/2018-07-10_23h00m32s
                                                                                                                                                    ^^^^^^^ birth
```
2018-07-10 was 988 days ago from now.
987 days (all expected):
```
~/git/poudriere6 # find /poudriere/data/logs/bulk/exp-head-i386-commit-test/*-*-*_*s -mindepth 0 -maxdepth 0 -type d -Btime +987d
/poudriere/data/logs/bulk/exp-head-i386-commit-test/2018-04-20_14h21m43s
/poudriere/data/logs/bulk/exp-head-i386-commit-test/2018-07-09_13h41m16s
/poudriere/data/logs/bulk/exp-head-i386-commit-test/2018-07-09_14h24m21s
/poudriere/data/logs/bulk/exp-head-i386-commit-test/2018-07-10_14h06m35s
/poudriere/data/logs/bulk/exp-head-i386-commit-test/2018-07-10_14h28m45s
/poudriere/data/logs/bulk/exp-head-i386-commit-test/2018-07-10_23h00m32s
```
```
~/git/poudriere6 # ./poudriere logclean -j exp-head-i386 -p commit -z test -n 987
[00:00:00] Looking for builds older than 987 days in /poudriere/data/logs/bulk (filtered)... done
[00:00:00] Calculating size for found files... done
[00:00:00] These builds older than 987 days in /poudriere/data/logs/bulk (filtered) will be deleted:
exp-head-i386-commit-test/2018-04-20_14h21m43s
exp-head-i386-commit-test/2018-07-09_13h41m16s
exp-head-i386-commit-test/2018-07-09_14h24m21s
exp-head-i386-commit-test/2018-07-10_14h06m35s
exp-head-i386-commit-test/2018-07-10_14h28m45s
exp-head-i386-commit-test/2018-07-10_23h00m32s
[00:00:00] Removing these 6 builds older than 987 days in /poudriere/data/logs/bulk (filtered) will free: 7.77 MiB
[00:00:00] Dry run: not cleaning anything.
[00:00:00] Looking for detached latest-per-pkg logfiles in /poudriere/data/logs/bulk (no filter)... done
[00:00:02] No detached latest-per-pkg logfiles in /poudriere/data/logs/bulk (no filter) to cleanup
[00:00:02] [Dry Run] Would remove broken legacy latest-per-pkg symlinks (no filter)...
[00:00:02] [Dry Run] Would remove empty latest-per-pkg directories (no filter)...
```

988 days (3 expected):
```
~/git/poudriere6 # find /poudriere/data/logs/bulk/exp-head-i386-commit-test/*-*-*_*s -mindepth 0 -maxdepth 0 -type d -Btime +988d
/poudriere/data/logs/bulk/exp-head-i386-commit-test/2018-04-20_14h21m43s
/poudriere/data/logs/bulk/exp-head-i386-commit-test/2018-07-09_13h41m16s
/poudriere/data/logs/bulk/exp-head-i386-commit-test/2018-07-09_14h24m21s
```
```
~/git/poudriere6 # ./poudriere logclean -j exp-head-i386 -p commit -z test -n 988
[00:00:00] Looking for builds older than 988 days in /poudriere/data/logs/bulk (filtered)... done
[00:00:00] Calculating size for found files... done
[00:00:00] These builds older than 988 days in /poudriere/data/logs/bulk (filtered) will be deleted:
exp-head-i386-commit-test/2018-04-20_14h21m43s
exp-head-i386-commit-test/2018-07-09_13h41m16s
exp-head-i386-commit-test/2018-07-09_14h24m21s
[00:00:00] Removing these 3 builds older than 988 days in /poudriere/data/logs/bulk (filtered) will free: 7.02 MiB
[00:00:00] Dry run: not cleaning anything.
[00:00:00] Looking for detached latest-per-pkg logfiles in /poudriere/data/logs/bulk (no filter)... done
[00:00:01] No detached latest-per-pkg logfiles in /poudriere/data/logs/bulk (no filter) to cleanup
[00:00:01] [Dry Run] Would remove broken legacy latest-per-pkg symlinks (no filter)...
[00:00:01] [Dry Run] Would remove empty latest-per-pkg directories (no filter)...
```

If this happens again we'll need some `stat` output on the dirs to see the birthtime.
